lib/_stream_duplex.js

```diff
@@ -70,63 +70,60 @@ function Duplex(options) {
-  // userland will fail
-  enumerable: false,
-  get() {
-    return this._writableState && this._writableState.highWaterMark;
-  }
-});
+ObjectDefineProperties(Duplex.prototype, {
 
-ObjectDefineProperty(Duplex.prototype, 'writableBuffer', {
-  // Making it explicit this property is not enumerable
-  // because otherwise some prototype manipulation in
-  // userland will fail
-  enumerable: false,
-  get: function() {
-    return this._writableState && this._writableState.getBuffer();
-  }
-});
+  destroyed: {
+    get() {
+      if (this._readableState === undefined ||
+        this._writableState === undefined) {
+        return false;
+      }
+      return this._readableState.destroyed && this._writableState.destroyed;
+    },
+    set(value) {
+      // Backward compatibility, the user is explicitly
+      // managing destroyed
+      if (this._readableState && this._writableState) {
+        this._readableState.destroyed = value;
+        this._writableState.destroyed = value;
+      }
+    }
+  },
 
-ObjectDefineProperty(Duplex.prototype, 'writableLength', {
-  // Making it explicit this property is not enumerable
-  // because otherwise some prototype manipulation in
-  // userland will fail
-  enumerable: false,
-  get() {
-    return this._writableState && this._writableState.length;
-  }
-});
+  writableHighWaterMark: {
+    get() {
+      return this._writableState && this._writableState.highWaterMark;
+    }
+  },
 
-ObjectDefineProperty(Duplex.prototype, 'writableFinished', {
-  // Making it explicit this property is not enumerable
-  // because otherwise some prototype manipulation in
-  // userland will fail
-  enumerable: false,
-  get() {
-    return this._writableState ? this._writableState.finished : false;
-  }
-});
+  writableBuffer: {
+    get() {
+      return this._writableState && this._writableState.getBuffer();
+    }
+  },
 
-ObjectDefineProperty(Duplex.prototype, 'writableCorked', {
-  // Making it explicit this property is not enumerable
-  // because otherwise some prototype manipulation in
-  // userland will fail
-  enumerable: false,
-  get() {
-    return this._writableState ? this._writableState.corked : 0;
-  }
-});
+  writableLength: {
+    get() {
+      return this._writableState && this._writableState.length;
+    }
+  },
 
-ObjectDefineProperty(Duplex.prototype, 'writableEnded', {
-  // Making it explicit this property is not enumerable
-  // because otherwise some prototype manipulation in
-  // userland will fail
-  enumerable: false,
-  get() {
-    return this._writableState ? this._writableState.ending : false;
+  writableFinished: {
+    get() {
+      return this._writableState ? this._writableState.finished : false;
+    }
+  },
+
+  writableCorked: {
+    get() {
+      return this._writableState ? this._writableState.corked : 0;
+    }
+  },
+
+  writableEnded: {
+    get() {
+      return this._writableState ? this._writableState.ending : false;
+    }
```

Comment:
Are we losing `enumerable: false`?

Reply:
@mcollina: The default value of `enumerable` is `false`. So we had just dropped it.